### DISCUSSION TITLE
How to show TextFieldType="TextFieldType.Password" in FluentUI TextField

### DISCUSSION THREAD

**Discussion by dosni** [microsoft/fluentui-blazor#1122] [General]
How do I show password in FluentUI TextField  with TextFieldType="TextFieldType.Password" ?

**Comment by vnbaaij** [comment]
Not sure what you mean here. You want to have the password reveal shown, like this?
![image](https://github.com/microsoft/fluentui-blazor/assets/1761079/56dce8dd-39f9-4110-a8aa-4fd178cfff08)
Because that is already working (see https://www.fluentui-blazor.net/TextField#password)

**Comment by dosni** [comment]
Yes I want to reveal it.  I dont have that icon on the right ?  I am using 4.20 on Net 8.

I can show it using code ..
  <FluentTextField Appearance="FluentInputAppearance.Outline" TextFieldType="@textPassword" style="width: 100%;" @bind-Value="user.Password" Placeholder="Password">
      <FluentIcon Slot="end" Icon="Icons.Regular.Size20.Password" Color="Color.Accent" OnClick="ShowPassword" />
  </FluentTextField>

@code{
 private TextFieldType textPassword { get; set; } = TextFieldType.Password;
    private void ShowPassword()
    {
        textPassword = textPassword == TextFieldType.Password ? TextFieldType.Text : TextFieldType.Password;
        
    }

}

**Reply by vnbaaij** [reply]
The reveal button is only showed when the text field is being filled in. Once the field loses focus and you navigate back to it, the button is not shown anymore. This is not specific to the library but is a general html (Chromium?) thing.

![password](https://github.com/microsoft/fluentui-blazor/assets/1761079/a982aad6-3175-4355-b944-090fd1619e87)

**Reply by vnbaaij** [reply]
Your code works as an option as well

**Comment by dosni** [comment]
![image](https://github.com/microsoft/fluentui-blazor/assets/87328342/8a7c22a3-c164-4111-b1f6-313063c9545d)

Why I can not see it on my Screen ?

**Reply by vnbaaij** [reply]
See above ☝️

**Reply by dosni** [reply]
I tried but not working for me.  I dont know what is wrong.  It is strage !. 

But it is okay. I just need to write some code to show and to hide. Thank you.

**Reply by svrooij** [reply]
@dosni this seems to be a browser feature. And @vnbaaij is probably testing in Edge (works). On our end it does not work in Chrome.
On other browsers you might have to trick the browser into showing the password by [changing the type](https://www.w3schools.com/howto/howto_js_toggle_password.asp) to `input`
